hilti/toolchain/include/ast/id.h

```diff
@@ -33,7 +33,7 @@ class ID : public detail::IDBase<ID> {
      * @param s string to construct from
      * @param meta meta data to attach
      */
-    explicit ID(std::string s, Meta meta) : IDBase(std::move(s)), _meta(std::move(meta)) {}
+    explicit ID(std::string_view s, Meta meta) : IDBase(s), _meta(std::move(meta)) {}
```

Comment:
With this overload available we should be able to get rid of the ctr from `const char*` above which now seems unused.

Reply:
That doesn't directly work unfortunately. For one, the `string_view` overload is explicit but the `const char *` isn't. That follows how it used to be and was on purposes (to allow passing string literals as IDs, but prevent string variables from implicit conversion). We could change that, however I just tried that and I'm still getting problems without the `const char*` overload where things don't convert over. Not sure why honestly. I'll leave as is for now.